### PR TITLE
Add detection logging messages at info level

### PR DESCRIPTION
**Pull request by anthonydahanne**
*Log messages were checked during testing*

```
=== RUN   TestUnit/gradle/Detect/fails_without_build.gradle_or_build.gradle.kts
SKIPPED: No plans could be resolved
=== RUN   TestUnit/gradle/Detect/fails_without_configured_build_file
SKIPPED: BP_GRADLE_BUILD_FILE was specified but /var/folders/rl/bdnmjrz16bz1wn7dblwvxmpc0000gq/T/gradle2621248551/var/folders/rl/bdnmjrz16bz1wn7dblwvxmpc0000gq/T/gradle2621248551/no-such-build.gradle could not be found
```

## Summary
This PR adds additional logging when Detect returns false for this buildpack. This is to provide more information about why the detection failed for this buildpack when verbose/debug logging is enabled. 

As of lifecycle v1.16.0, if the detect phase fails as a whole, log messages are printed at info level by default, without the need for the verbose flag.

## Use Cases
A `pack build` command with the `--verbose` flag set will show these log statement if detection fails. If Detect succeeds, the output is not shown unless `--verbose` is set.

## Checklist

* [X] I have viewed, signed, and submitted the Contributor License Agreement.
* [X] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [X] I have added an integration test, if necessary.
* [X] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [X] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Comment by dmikusa** [comment]
Looks good. Same comment as the image-labels PR regarding `libpak.NewConfigurationResolver`. Thoughts?

>As of lifecycle v1.16.0, if the detect phase fails as a whole, log messages are printed at info level by default, without the need for the verbose flag.

Neat, I learned something new. Thanks

**Comment by anthonydahanne** [comment]
> Looks good. Same comment as the image-labels PR regarding libpak.NewConfigurationResolver. Thoughts?

If I add the bard logger to `libpak.NewConfigurationResolver`, in case of a detect problem:

```
pack build -p ~/workspaces/paketo-buildpacks/samples/java/gradle --env BP_JVM_VERSION=17  gradle:detect-logging \
  -b paketo-buildpacks/ca-certificates \
  -b paketo-buildpacks/bellsoft-liberica \
  -b paketo-buildpacks/syft \
  -b /Users/anthonyd2/workspaces/paketo-buildpacks/gradle \
  -b paketo-buildpacks/executable-jar \
  -b paketo-buildpacks/apache-tomcat \
  -b paketo-buildpacks/apache-tomee \
  -b paketo-buildpacks/liberty \
  -b paketo-buildpacks/dist-zip \
  -b paketo-buildpacks/spring-boot --clear-cache --env BP_GRADLE_BUILD_FILE=nope.gradle
[...]
===> DETECTING
======== Output: paketo-buildpacks/gradle@{{.version}} ========
  Build Configuration:
    $BP_EXCLUDE_FILES                                            colon separated list of glob patterns, matched source files are removed
    $BP_GRADLE_ADDITIONAL_BUILD_ARGUMENTS                        the additionnal arguments (appended to BP_GRADLE_BUILD_ARGUMENTS) to pass to Gradle
    $BP_GRADLE_BUILD_ARGUMENTS             --no-daemon assemble  the arguments to pass to Gradle
    $BP_GRADLE_BUILD_FILE                  nope.gradle           the location of the main build config file, relative to the application root
    $BP_GRADLE_BUILT_ARTIFACT              build/libs/*.[jw]ar   the built application artifact explicitly.  Supersedes $BP_GRADLE_BUILT_MODULE
    $BP_GRADLE_BUILT_MODULE                                      the module to find application artifact in
    $BP_GRADLE_INIT_SCRIPT_PATH                                  the path to a Gradle init script file
    $BP_INCLUDE_FILES                                            colon separated list of glob patterns, matched source files are included
    $BP_JAVA_INSTALL_NODE                  false                 whether to install Yarn/Node binaries based on the presence of a package.json or yarn.lock file
    $BP_NODE_PROJECT_PATH                                        configure a project subdirectory to look for `package.json` and `yarn.lock` files
SKIPPED: BP_GRADLE_BUILD_FILE was specified but /workspace/nope.gradle could not be found
```

in case of no issues:
```
pack build -p ~/workspaces/paketo-buildpacks/samples/java/gradle --env BP_JVM_VERSION=17  gradle:detect-logging \
  -b paketo-buildpacks/ca-certificates \
  -b paketo-buildpacks/bellsoft-liberica \
  -b paketo-buildpacks/syft \
  -b /Users/anthonyd2/workspaces/paketo-buildpacks/gradle \
  -b paketo-buildpacks/executable-jar \
  -b paketo-buildpacks/apache-tomcat \
  -b paketo-buildpacks/apache-tomee \
  -b paketo-buildpacks/liberty \
  -b paketo-buildpacks/dist-zip \
  -b paketo-buildpacks/spring-boot --clear-cache
[...]
===> DETECTING
paketo-buildpacks/ca-certificates   3.6.1
paketo-buildpacks/bellsoft-liberica 10.2.3
paketo-buildpacks/syft              1.30.0
paketo-buildpacks/gradle            {{.version}}
paketo-buildpacks/executable-jar    6.7.1
paketo-buildpacks/apache-tomcat     7.13.3
paketo-buildpacks/apache-tomee      1.7.1
paketo-buildpacks/liberty           3.7.1
paketo-buildpacks/dist-zip          5.6.1
paketo-buildpacks/spring-boot       5.25.0
```

and in case of no error, with verbose:
```
===> DETECTING
======== Output: paketo-buildpacks/gradle@{{.version}} ========
  Build Configuration:
    $BP_EXCLUDE_FILES                                            colon separated list of glob patterns, matched source files are removed
    $BP_GRADLE_ADDITIONAL_BUILD_ARGUMENTS                        the additionnal arguments (appended to BP_GRADLE_BUILD_ARGUMENTS) to pass to Gradle
    $BP_GRADLE_BUILD_ARGUMENTS             --no-daemon assemble  the arguments to pass to Gradle
    $BP_GRADLE_BUILD_FILE                                        the location of the main build config file, relative to the application root
    $BP_GRADLE_BUILT_ARTIFACT              build/libs/*.[jw]ar   the built application artifact explicitly.  Supersedes $BP_GRADLE_BUILT_MODULE
    $BP_GRADLE_BUILT_MODULE                                      the module to find application artifact in
    $BP_GRADLE_INIT_SCRIPT_PATH                                  the path to a Gradle init script file
    $BP_INCLUDE_FILES                                            colon separated list of glob patterns, matched source files are included
    $BP_JAVA_INSTALL_NODE                  false                 whether to install Yarn/Node binaries based on the presence of a package.json or yarn.lock file
    $BP_NODE_PROJECT_PATH                                        configure a project subdirectory to look for `package.json` and `yarn.lock` files
======== Results ========
pass: paketo-buildpacks/ca-certificates@3.6.1
pass: paketo-buildpacks/bellsoft-liberica@10.2.3
pass: paketo-buildpacks/syft@1.30.0
pass: paketo-buildpacks/gradle@{{.version}}
pass: paketo-buildpacks/executable-jar@6.7.1
pass: paketo-buildpacks/apache-tomcat@7.13.3
pass: paketo-buildpacks/apache-tomee@1.7.1
pass: paketo-buildpacks/liberty@3.7.1
pass: paketo-buildpacks/dist-zip@5.6.1
pass: paketo-buildpacks/spring-boot@5.25.0
Resolving plan... (try #1)
fail: paketo-buildpacks/ca-certificates@3.6.1 provides unused ca-certificates
Resolving plan... (try #2)
fail: paketo-buildpacks/executable-jar@6.7.1 requires jre
Resolving plan... (try #3)
fail: paketo-buildpacks/ca-certificates@3.6.1 provides unused ca-certificates
Resolving plan... (try #4)
fail: paketo-buildpacks/executable-jar@6.7.1 requires jre
Resolving plan... (try #5)
fail: paketo-buildpacks/gradle@{{.version}} requires jdk
Resolving plan... (try #6)
fail: paketo-buildpacks/bellsoft-liberica@10.2.3 provides unused native-image-builder
Resolving plan... (try #7)
fail: paketo-buildpacks/executable-jar@6.7.1 requires jre
Resolving plan... (try #8)
paketo-buildpacks/ca-certificates   3.6.1
paketo-buildpacks/bellsoft-liberica 10.2.3
paketo-buildpacks/syft              1.30.0
paketo-buildpacks/gradle            {{.version}}
paketo-buildpacks/executable-jar    6.7.1
paketo-buildpacks/apache-tomcat     7.13.3
paketo-buildpacks/apache-tomee      1.7.1
paketo-buildpacks/liberty           3.7.1
paketo-buildpacks/dist-zip          5.6.1
paketo-buildpacks/spring-boot       5.25.0
```

so all in all, I think it's a pretty good addition; I was fearing it would added even with no error, no verbose, but no, it's only in verbose or error; just like the logging I added.

so yeah, updating the PR !

File: scripts/build.sh [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 set -euo pipefail
 
-GOOS="linux" go build -ldflags='-s -w' -tags osusergo -o bin/main github.com/paketo-buildpacks/gradle/v6/cmd/main
+GOARCH="amd64" GOOS="linux" go build -ldflags='-s -w' -tags osusergo -o bin/main github.com/paketo-buildpacks/gradle/v6/cmd/main
```
Comment (dmikusa) [inline scripts/build.sh:5]: Was this intentional? We're going to need to figure out how to handle architecture soon, but I'd rather not start adding bits in until we've got a plan so we can apply the same across all of the buildpacks. Unless there's a specific need for this now. Thanks

Comment (anthonydahanne) [reply scripts/build.sh:5]: no sorry, it was not.

just that I use that on my machine, and forgot to remove it committing

**Review by pivotal-david-osullivan — APPROVED** [review]

**Pull request merged**